fastify.js

```diff
@@ -103,7 +113,10 @@ function build (options) {
   fastify.hasContentTypeParser = hasContentTypeParser
   fastify._contentTypeParser = new ContentTypeParser()
 
-  fastify.schemaCompiler = schemaCompiler
+  // expose ajv instance
+  fastify.ajv = ajv
```

Comment:
Why expose `Ajv`?

Reply:
Because ajv has a great API (add keywords, add plugin etc..)

Reply:
I'm not sure about this.
If you need that kind of configuration you could use `setSchemaCompiler` and you are done.
If in the future we choose to remove `Ajv` in favor of another library this could be a breaking change.

Reply:
When we provide specific customization for the ajv it would be a breaking change too.

Reply:
Yes, but you will have `setSchemaCompiler` to easily fix the issue.
IMHO one thing is a generic api that supports every library, another thing is expose an object of an internal library that could change during time.
Again, if you need that kind of customization `setSchemaCompiler` is the right choice. We should not provide too many api, or the result will be confusing :)

Reply:
Ok, I understand your point but then it makes no sense to me to provide a way to customize the ajv instance when I can't work with as what it is. Then I prefer to only provide the `setSchemaCompiler` solution.

Reply:
Should we close the PR ?

Reply:
I am a bit lost. Adding some custom options for `ajv` is a good thing, so don't close the PR.

I'm 👎  on exposing it. Exposing ajv API would force us to bump semver-major if we upgrade the ajv semver-major, while as things are now the parts ajv API that we are exposing are extremely thin.

Reply:
That's the list of https://github.com/epoberezkin/ajv#options options which ajv provides. We will definitely be careful about updates. Should we provide a limited set?

```
 removeAdditional: false,
  useDefaults:      false,
  coerceTypes:      false,
```
would match the most use cases.